pex/pex.py

```diff
@@ -415,6 +410,40 @@ def _execute(self):
       TRACER.log('No entry point specified, dropping into interpreter')
       return self.execute_interpreter()
 
+  @classmethod
+  @contextmanager
```

Comment:
any point to keeping this as a `contextmanager` if it doesn't snapshot and restore prior `sys.path` state? as a caller, I'd expect the contextmgr to undo any mutations vs purely apply them and do nothing upon context block exit.

Reply:
Nope. I did start off re-jiggering in finally until I realized it was not needed. I'll fix. The `patch_sys` has this problem too.

Reply:
Worth another look - I stripped the contextmanagers from this class. Only one tried to restore anything (`patch_pkg_resources`) and the restoration was not needed for the pex handling of teardown.